x/mongo/driver/topology/rtt_monitor.go

```diff
@@ -323,6 +328,6 @@ func (r *rttMonitor) Stats() string {
 	}
 
 	return fmt.Sprintf(`Round-trip-time monitor statistics:`+"\n"+
-		`average RTT: %v, minimum RTT: %v, 90th percentile RTT: %v, standard dev: %v`+"\n",
-		time.Duration(avg), r.minRTT, r.rtt90, time.Duration(stdDev))
+		`average RTT: %v, minimum RTT: %v, standard dev: %v`+"\n",
+		time.Duration(avg), r.minRTT, time.Duration(stdDev))
```

Comment:
Can we calculate stdDev using the `movingMin` values instead of `samples`?

Reply:
There is no mention in the specifications on propagating the standard deviation of RTT samples, so I think we can use the full set of samples or the `movingMin` window. 

I'm assuming that we provide this information for users to determine if the stability of the connection could potentially be responsible for a timeout error. That is, _how close are our the RTTs to the mean?_ Basing this information on the `movingMin`, a a 10-element window of all of the samples, will only tell us how close that window is to it's own mean. Furthermore, there is no obvious way to calculate the overall standard deviation as an aggregation of these windows since the windows themselves are not disjoint. If the goal is to give the user insights on the variability or dispersion of the RTT values, then not using the full sample set would be a mistake.

Reply:
The stddev was indeed added to give users (and us) an idea of the distribution of RTT measurements when the RTT logic short-circuits an operation. That was more important when we were using the 90th percentile RTT, which is more sensitive to widely distributed RTT values than the minimum. However, since we're using minimum and only keeping 10 samples, I think we should include those 10 samples in the error message instead of deriving stddev.

Keep in mind that `samples` isn't a complete set of RTT samples either, it's a moving window of 5 minutes of samples.

Reply:
Calculating the standard deviation of such a small subset (~2% in the worst case) and propagating it to the user is a mistake and would very likely not be an accurate representation of connection (in)stability.

If we have to get rid of `samples` and quantify connection stability, we could create a type to hold the sum of standard deviations calculated evert minRTTSamplesForMovingMin-th call to `appendMovingMin`. We would need to keep another type to keep track of calls to `appendMovingMin` Then we could provide an average standard deviation, something like [this](https://stats.stackexchange.com/a/276035):

```go
const maxRTTSamplesForMovingMin = 10

type rttMonitor struct {
	varSum float64

	callsToAppendMovingMin int
	movingMin              *list.List
}

func (rtt *rttMonitor) appendMovingMin(t time.Duration) {
	defer func() { rtt.callsToAppendMovingMin++ }()

	if rtt.movingMin == nil || t < 0 {
		return
	}

	if rtt.movingMin.Len() == maxRTTSamplesForMovingMin {
		rtt.movingMin.Remove(rtt.movingMin.Front())
	}

	rtt.movingMin.PushBack(t)

	// Collect a sum of variances ever 10 calls, ignoring the fist call.
	if rtt.callsToAppendMovingMin > 1 && rtt.callsToAppendMovingMin%10 == 0 {
		// Piecewise update latest variance, varSum
	}
}
```

We could also just qualify using the last ten samples:

```go
	return fmt.Sprintf(`Round-trip-time monitor statistics:`+"\n"+
		`average RTT: %v, minimum RTT: %v, %v, standard dev of last 10 samples: %v`+"\n",
		time.Duration(avg), r.minRTT, r.rtt90, time.Duration(stdDev))
```

Reply:
I do think we should remove `samples` since it adds a lot of complexity for only giving us stddev. That also allows us to stop using the "github.com/montanaflynn/stats" package and remove it from our dependencies (after removing it from the `benchmark` package).

Your plan to calculate the average stddev sounds great! Should we use average or a moving average like EWMA?

Reply:
MSD would be more sensitive to dispersion, which is good. This would just be the average standard deviations of the $S_{10}$ moving average defined [here](https://mathworld.wolfram.com/MovingAverage.html). Which would result in a Nx1 vector of standard deviations, like [this](https://www.mathworks.com/help/matlab/ref/movstd.html#d126e1035934). This would be the only change required IIUC:

```go
	if rtt.callsToAppendMovingMin >= maxRTTSamplesForMovingMin {
		// Update latest variance, varSum
	}
```

Here is a working example, you roughly confirm the hypothesis by adjusting the `rang` variable which should grow larger by a factor of itself: https://go.dev/play/p/s54Z4BQWPCR